unit_tests/src/test_zone_administration.cpp

```diff
@@ -0,0 +1,126 @@
+    }
+
+    SECTION("create zone with connection information and comment")
+    {
+        REQUIRE_NOTHROW(adm::client::add_zone(conn, test_zone_name_1, {conn_info_1, comment_1}));
+
+        const auto zone_info = adm::client::zone_info(conn, test_zone_name_1);
+        REQUIRE(zone_info.has_value());
+        CHECK(zone_info->type == adm::zone_type::remote);
+        CHECK(zone_info->name == test_zone_name_1);
+        CHECK(zone_info->connection_info == conn_info_1);
+        CHECK(zone_info->comment == comment_1);
+    }
+
+    // Make sure to remove zone(s) on completion (this covers exceptions too).
+    irods::at_scope_exit remove_test_zones{[&conn, &test_zone_name_1, &test_zone_name_2] {
+        if (adm::client::zone_exists(conn, test_zone_name_1)) {
+            adm::client::remove_zone(conn, test_zone_name_1);
+        }
+
+        if (adm::client::zone_exists(conn, test_zone_name_2)) {
+            adm::client::remove_zone(conn, test_zone_name_2);
+        }
+    }};
+
+    // Show that zones can be renamed.
+    CHECK_FALSE(adm::client::zone_exists(conn, test_zone_name_2));
+    CHECK_NOTHROW(adm::client::modify_zone(conn, test_zone_name_1, adm::zone_name_property{test_zone_name_2}));
+    CHECK_FALSE(adm::client::zone_exists(conn, test_zone_name_1));
+
+    CHECK_NOTHROW(adm::client::modify_zone(conn, test_zone_name_2, adm::zone_name_property{test_zone_name_1}));
+    CHECK(adm::client::zone_exists(conn, test_zone_name_1));
+
+    // Show that the connection information can be adjusted.
+    auto zone_info = adm::client::zone_info(conn, test_zone_name_1);
+    REQUIRE(zone_info.has_value());
+    CHECK(zone_info->type == adm::zone_type::remote);
+    CHECK(zone_info->name == test_zone_name_1);
+
+    const auto* const conn_info_2 = "unit-test.example.org:1247";
+    CHECK_NOTHROW(adm::client::modify_zone(conn, test_zone_name_1, adm::connection_info_property{conn_info_2}));
+
+    zone_info = adm::client::zone_info(conn, test_zone_name_1);
+    REQUIRE(zone_info.has_value());
+    CHECK(zone_info->type == adm::zone_type::remote);
+    CHECK(zone_info->name == test_zone_name_1);
+    CHECK(zone_info->connection_info == conn_info_2);
+
+    // Show that the comments can be adjusted.
+    const auto* const comment_2 = "updated comment.";
+    CHECK_NOTHROW(adm::client::modify_zone(conn, test_zone_name_1, adm::comment_property{comment_2}));
+
+    zone_info = adm::client::zone_info(conn, test_zone_name_1);
+    REQUIRE(zone_info.has_value());
+    CHECK(zone_info->type == adm::zone_type::remote);
+    CHECK(zone_info->name == test_zone_name_1);
+    CHECK(zone_info->connection_info == conn_info_2);
+    CHECK(zone_info->comment == comment_2);
+
+    // Show that the handling of the zone collection in regards to strict-acls can be modified.
+    const adm::zone_collection_acl_property prop{adm::zone_collection_acl::read, "rods"};
+    CHECK_NOTHROW(adm::client::modify_zone(conn, test_zone_name_1, prop));
+
```

Comment:
Only suggestion I have is to maybe add a test or two that exercises the error cases, i.e. `CHECK_THROWS`.

Reply:
I've added some basic tests for error cases.